src/toolbars/ToolBarButtons.cpp

```diff
@@ -0,0 +1,125 @@
+#include <utility>
+#include <wx/app.h>
+
+#include "ToolBarButtons.h"
+
+#include "AllThemeResources.h"
+#include "../widgets/AButton.h"
+#include "Project.h"
+
+#include "../commands/CommandContext.h"
+#include "../commands/CommandManager.h"
+#include "../commands/CommandDispatch.h"
+
+// flags so 1,2,4,8 etc.
+enum {
+   TBActTooltips = 1,
+   TBActEnableDisable = 2,
+};
+
+ToolBarButtons::ToolBarButtons(ToolBar *const parent, AudacityProject & project, ButtonList buttonList, int size, int firstButtonId)
+   : mParent(parent)
+   , mProject(project)
+   , mButtonList(std::move(buttonList))
+   , mFirstButtonId(firstButtonId)
+{
+   mButtons.resize(size);
+}
+
+void ToolBarButtons::OnButton(wxCommandEvent & event)
+{
+   int id = event.GetId() - mFirstButtonId;
+
+   // Be sure the pop-up happens even if there are exceptions, except for buttons which toggle.
+   auto cleanup = finally( [&] { mButtons[id]->InteractionOver(); });
+
+   auto &cm = CommandManager::Get( mProject );
+
+   auto flags = MenuManager::Get(mProject).GetUpdateFlags();
+   const CommandContext context( mProject );
+
+   ::HandleTextualCommand( cm, mButtonList[id].commandName, context, flags, false );
+
+#if defined(__WXMAC__)
+   // Bug 2402
+   // LLL: It seems that on the Mac the IDLE events are processed
+   //      differently than on Windows/GTK and the AdornedRulerPanel's
+   //      OnPaint() method gets called sooner that expected. This is
+   //      evident when zooming from this toolbar only. When zooming from
+   //      the Menu or from keyboard ommand, the zooming works correctly.
+   wxTheApp->ProcessIdle();
+#endif
+}
+
+AButton* ToolBarButtons::CreateButton(teBmps eEnabledUp, teBmps eEnabledDown, teBmps eDisabled, int thisButtonId, const TranslatableString& label, bool toggle)
+{
+   AButton *&r = mButtons[thisButtonId];
+
+   r = ToolBar::MakeButton(mParent,
+      bmpRecoloredUpSmall, bmpRecoloredDownSmall, bmpRecoloredUpHiliteSmall, bmpRecoloredHiliteSmall,
+      eEnabledUp, eEnabledDown, eDisabled,
+      wxWindowID(thisButtonId + mFirstButtonId),
```

Comment:
Two integers are passed into this function, just so they can be added.

Why not simplify, passing in just the sum?

Reply:
OK, a later commit changes this

Reply:
I am not quite sure what do you mean

Reply:
I was reviewing the commits sequentially.  My comment applies to a step of the code transformation that is not the last.  The final version uses only one id argument.